worker/worker.go

```diff
@@ -76,7 +79,7 @@ func (w *Worker) run() {
 					break
 				}
 				w.log.Debugf("[worker] receive task: %v", task)
-				safe.Go(func() {
+				err = w.pool.SubmitFn(time.Second*10, func() {
```

Comment:
What's the purpose of using timeout?

Reply:
The idea is that by using a timeout to give up this submit try to reduce the possibility of duplicate delivery.

The invisible queue retries messages that exceed the invisible_timeout, so the next call of queue.Get may get the same message if it was blocking long enough.